include/zephyr/net_buf.h

```diff
@@ -1832,6 +1833,72 @@ static inline void net_buf_add_be64(struct net_buf *buf, uint64_t val)
 	net_buf_simple_add_be64(&buf->b, val);
 }
 
+/**
+ * @brief Add a character string to the end of the buffer
+ *
+ * Adds a @c \\0 terminated string to buffer if it fits.
+ * The terminating @c \\0 itself is not added.
+ *
+ * See @ref net_buf_simple_to_str to access the data as a regular C string.
+ *
+ * @param buf Buffer to update.
+ * @param s String to be added.
+ *
+ * @return 0 if the formatted string could fit.
+ * @return -ENOMEM if the buffer was too short for the formatted string.
+ */
+int net_buf_simple_add_str(struct net_buf_simple *buf, const char *s);
```

Comment:
There might be a need for a function where you can specify the length too.
```
net_buf_simple_add_str_len(buf, str, len)
```
And the function would only copy max len bytes.

Please also have slightly longer variable names so `str` instead of `s` just to make sure that it means a string.

Reply:
Done as per latest push. Thank you for your reviews!